skll/config.py

```diff
@@ -338,6 +353,17 @@ def _parse_config_file(config_path):
     custom_learner_path = _locate_file(config.get("Input", "custom_learner_path"),
                                        config_dir)
 
+    # Check that the learners are valid learner types (or that they are custom
+    # learners, i.e., ending in ".py")
```

Comment:
This check seems a bit overly complex when you consider that we already have [these checks](https://github.com/EducationalTestingService/skll/blob/master/skll/learner.py#L199-L205) for custom learners and [this](https://github.com/EducationalTestingService/skll/blob/master/skll/learner.py#L546) will raise a `KeyError` if a learner is not valid.

Is the thought process that we want to raise an exception here before we get there?  If so, it seems like there's got to be a simpler way to do this other than needing this whole `_LEARNER_NAMES_TO_CLASSES` thing.  You could probably even just look in just do:

```py
unrecognized_learner_names = {learner_name for learner_name in learner_names 
                              if 'skll.learner.{}'.format(learner_name) not in globals()}
```